paddle/fluid/framework/program_processing.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/fluid/framework/program_processing.h"
+#include "paddle/fluid/framework/block_desc.h"
+
+namespace paddle {
+namespace framework {
+
+void ProgramProcessor::GetInputsOutputsInBlock(
+    ProgramDesc *program, const BlockDesc &current_block,
+    std::set<std::string> *inner_inputs, std::set<std::string> *inner_outputs) {
+  /* Find inputs and outputs in current control flow block.
+  :param program: Program of current control flow block.
+  :param current_block: Current control flow block.
+  :param inner_inputs: Input var name of ops in current block.
+  :param inner_outputs: Output var name of ops in current block. */
+
+  // Step1: update inner_inputs and inner_outputs
+  // NOTE: Here assumes that all variables are input or output of Ops,
+
+  std::set<std::string> removed_inner_inputs;
+
+  for (OpDesc *op : current_block.AllOps()) {
+    for (auto iname : op->InputNames())
+      for (auto in_var_name : op->Inputs().find(iname)->second) {
+        if ((*inner_outputs).find(in_var_name) == (*inner_outputs).end())
+          (*inner_inputs).insert(in_var_name);
```

Comment:
`inner_inputs->insert(in_var_name) ` is a common coding pattern in this case. Same for other similar places:

`(*pointer).` can be replaced by `pointer->`

Reply:
Done.